### PR TITLE
Final Project Mode Corrections/Changes

### PR DESCRIPTION
**Pull request by gustavomm19**
https://github.com/4GeeksAcademy/About-4Geeks-Academy/issues/4251

### CODE REVIEW

**Review by tommygonzaleza — DISMISSED** [review]

File: breathecode/assignments/views.py [hunk 1]
```diff
@@ -128,6 +128,61 @@ def sync_cohort_tasks_view(request, cohort_id=None):
...
+        }
+
+        file = request.data.get('file')
+        slugs = []
+
+        for index in range(0, len(files)):
+            file = files[index]
+            if file.content_type not in MIME_ALLOW:
+                raise ValidationException(
+                    f'You can upload only files on the following formats: {",".join(MIME_ALLOW)}')
```
Comment (jefer94) [inline breathecode/assignments/views.py:153]: Only allow mime types of images

File: breathecode/assignments/views.py [hunk 2]
```diff
@@ -128,6 +128,61 @@ def sync_cohort_tasks_view(request, cohort_id=None):
...
+            file = files[index]
+            name = names[index] if len(names) else file.name
+            file_bytes = file.read()
+            hash = hashlib.sha256(file_bytes).hexdigest()
+            slug = slugify(name)
+
+            slugs.append(slug)
+            data = {
+                'hash': hash,
+                'slug': slug,
```
Comment (jefer94) [inline breathecode/assignments/views.py:165]: Remove this

File: breathecode/assignments/views.py [hunk 3]
```diff
@@ -128,6 +128,61 @@ def sync_cohort_tasks_view(request, cohort_id=None):
...
+                'mime': file.content_type,
+                'name': name,
+                'categories': [],
```
Comment (jefer94) [inline breathecode/assignments/views.py:168]: Remove this

File: breathecode/assignments/views.py [hunk 4]
```diff
@@ -128,6 +128,61 @@ def sync_cohort_tasks_view(request, cohort_id=None):
...
+            file_bytes = file.read()
+            hash = hashlib.sha256(file_bytes).hexdigest()
+            slug = slugify(name)
+
+            slugs.append(slug)
+            data = {
+                'hash': hash,
+                'slug': slug,
+                'mime': file.content_type,
+                'name': name,
```
Comment (jefer94) [inline breathecode/assignments/views.py:167]: Remove this

File: breathecode/assignments/views.py [hunk 5]
```diff
@@ -128,6 +128,61 @@ def sync_cohort_tasks_view(request, cohort_id=None):
...
+                'categories': [],
+            }
+
+            # upload file section
+            storage = Storage()
+            cloud_file = storage.file(USER_ASSIGNMENTS_BUCKET, hash)
+            cloud_file.upload(file, content_type=file.content_type)
+            data['url'] = cloud_file.url()
+            data['thumbnail'] = data['url'] + '-thumbnail'
```
Comment (jefer94) [inline breathecode/assignments/views.py:176]: If you don't use the thumbnail, remove this

File: breathecode/assignments/views.py [hunk 6]
```diff
@@ -189,6 +244,10 @@ def post(self, request, user_id=None):
         if isinstance(request.data, list) == False:
             payload = [request.data]
 
+        members_set = set(payload[0]['members'])
```
Comment (jefer94) [inline breathecode/assignments/views.py:247]: `user_id` will be `request.user.id` in all cases

**Review by jefer94 — CHANGES_REQUESTED** [review]

**Comment by jefer94** [comment]
Check the corrections about the object that return the upload view

**Pull request merged**